src/torchcodec/decoders/_core/AVIOFileLikeContext.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "src/torchcodec/decoders/_core/AVIOFileLikeContext.h"
+#include <torch/types.h>
+
+namespace facebook::torchcodec {
+
+AVIOFileLikeContext::AVIOFileLikeContext(py::object fileLike)
+    : fileLike_{UniquePyObject(new py::object(fileLike))} {
```

Comment:
I'm struggling to parse

```Cpp
new py::object(fileLike))
```

We're allocating a new `py::object` from an existing `py::object`, which is `fileLike`? What kind of copy is involved here (if any?)

Reply:
Correct. We're invoking `py::object`'s [copy constructor](https://en.cppreference.com/w/cpp/language/copy_constructor) on `fileLike`. That resulting object is dynamically allocated on the heap, and its pointer is returned by `new`. The pointer to that object is then owned by `AVIOFileLikeContext::fileLike_`.

It may be instructive to contrast this to a stack allocated object, in statements that appear in normal code blocks:
```cpp
py::object allocatedOnStack(fileLike);
py::object* allocatedOnHeap = new py::object(fileLike);
```